### PR TITLE
Add analyser for eclExt data format in France

### PR DESCRIPTION
**Pull request by flacombe**
EclExt is a French national standard for street lighting data
http://cnig.gouv.fr/eclairage-exterieur-a25020.html

Starting with a file over Haute-Savoie published by Syane.

### CODE REVIEW

**Comment by frodrigo** [comment]
Please can you rename files:
- analyser_merge_eclext_FR.py to Analyser_Merge_Lit_FR_eclext.py
- analyser_merge_eclext_FR_Syane.py to analyser_merge_lit_FR_haute-savoie_syane.py

File: analysers/analyser_merge_eclext_FR.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+
+class Analyser_merge_Eclext_FR(Analyser_Merge_Point):
+    def __init__(self, config, source_url, dataset_name, source, srid, osmRef, logger = None):
+        Analyser_Merge_Point.__init__(self, config, logger)
+        self.def_class_missing_official(item = 8090, id = 11, level = 3, tags = ['merge', 'eclairage', 'eclext', 'fix:chair', 'fix:survey'],
+            title = T_('Street light not integrated'))
+        self.def_class_possible_merge(item = 8091, id = 13, level = 3, tags = ['merge', 'eclairage', 'eclext', 'fix:chair', 'fix:picture'],
+            title = T_('Street light integration suggestion'))
+        self.def_class_update_official(item = 8092, id = 14, level = 3, tags = ['merge', 'eclairage', 'eclext', 'fix:chair', 'fix:survey'],
```
Comment (frodrigo) [inline analysers/analyser_merge_eclext_FR.py:35]: Ids should go to Analyser_Merge_Eclext_FR_Syane

File: analysers/analyser_merge_eclext_FR.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+from modules.OsmoseTranslation import T_
+from dateutil.parser import parse
+from .Analyser_Merge import Analyser_Merge_Point, CSV, Load_XY, Conflate, Select, Mapping
+
+
+class Analyser_merge_Eclext_FR(Analyser_Merge_Point):
+    def __init__(self, config, source_url, dataset_name, source, srid, osmRef, logger = None):
+        Analyser_Merge_Point.__init__(self, config, logger)
+        self.def_class_missing_official(item = 8090, id = 11, level = 3, tags = ['merge', 'eclairage', 'eclext', 'fix:chair', 'fix:survey'],
```
Comment (frodrigo) [inline analysers/analyser_merge_eclext_FR.py:31]: Tag, only english OSM like keys.
- eclairage -> lit
- eclext -> to be removed

File: analysers/analyser_merge_eclext_FR.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+            title = T_('Street light not integrated'))
+        self.def_class_possible_merge(item = 8091, id = 13, level = 3, tags = ['merge', 'eclairage', 'eclext', 'fix:chair', 'fix:picture'],
+            title = T_('Street light integration suggestion'))
+        self.def_class_update_official(item = 8092, id = 14, level = 3, tags = ['merge', 'eclairage', 'eclext', 'fix:chair', 'fix:survey'],
+            title = T_('Street light update'))
+
+        def extract_light_method(res):
+            if res.get('typeSource') == 'FLUO':
```
Comment (frodrigo) [inline analysers/analyser_merge_eclext_FR.py:39]: You can just dict for this kind of thing.
```python
extract_light_method = {
  'FLUO':  'fluorescent'
}
```

File: analysers/analyser_merge_eclext_FR_Syane.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+from .analyser_merge_eclext_FR import Analyser_merge_Eclext_FR
+from .Analyser_Merge import Source
+
+
+class Analyser_Merge_Eclext_FR_Syane(Analyser_merge_Eclext_FR):
+    def __init__(self, config, logger = None):
+        Analyser_merge_Eclext_FR.__init__(self, config,
+            source_url='https://www.data.gouv.fr/fr/datasets/points-lumineux-declairage-public-exterieur-dont-le-syane-est-exploitant-haute-savoie/',
+            dataset_name='Points lumineux d\'éclairage public extérieur dont le Syane est exploitant (Haute-Savoie)',
+            source=Source(attribution='Syane',
```
Comment (frodrigo) [inline analysers/analyser_merge_eclext_FR_Syane.py:32]: Please, use a `dataGouv` source not a generic `Source`.

File: analysers/analyser_merge_eclext_FR.py (hunk 1 above)
Comment (flacombe) [reply analysers/analyser_merge_eclext_FR.py:35]: Is it necessary?

I was inspired by `analyser_merge_hydrants_FR.py`

File: analysers/analyser_merge_eclext_FR_Syane.py (hunk 4 above)
Comment (flacombe) [reply analysers/analyser_merge_eclext_FR_Syane.py]: Done, thanks

File: analysers/analyser_merge_eclext_FR.py (hunk 3 above)
Comment (flacombe) [reply analysers/analyser_merge_eclext_FR.py]: Done

File: analysers/analyser_merge_eclext_FR.py (hunk 1 above)
Comment (frodrigo) [reply analysers/analyser_merge_eclext_FR.py]: You split into two file to make reusable. So yes.

> I was inspired by analyser_merge_hydrants_FR.py

That bad. But in this case it works anyway.

File: analysers/Analyser_Merge_street_lamp_FR_eclext.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+            def mapper(res):
+                v = res.get(name)
+                if v:
+                    if v != "INCONNU":
+                        return v.lower()
+                    else:
+                        return None
+                else:
+                    return None
+            return mapper
```
Comment (frodrigo) [inline analysers/Analyser_Merge_street_lamp_FR_eclext.py:91]: Same, you just use a `dict`.

File: analysers/Analyser_Merge_street_lamp_FR_eclext.py [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    def __init__(self, config, source_url, dataset_name, source, srid, osmRef, logger = None):
+        Analyser_Merge_Point.__init__(self, config, logger)
+        self.def_class_missing_official(item = 8090, id = 11, level = 3, tags = ['merge', 'street_lamp', 'fix:chair', 'fix:survey'],
+            title = T_('Street light not integrated'))
+        self.def_class_possible_merge(item = 8091, id = 13, level = 3, tags = ['merge', 'street_lamp', 'fix:chair', 'fix:picture'],
+            title = T_('Street light integration suggestion'))
+        self.def_class_update_official(item = 8092, id = 14, level = 3, tags = ['merge', 'street_lamp', 'fix:chair', 'fix:survey'],
+            title = T_('Street light update'))
+
+        extract_light_method = {
```
Comment (frodrigo) [inline analysers/Analyser_Merge_street_lamp_FR_eclext.py:38]: Please, declare this and all the flowing directly into the class, not into __init__().

**Comment by frodrigo** [comment]
You also nee to add your new analyser to the configuration.

File: analysers/analyser_merge_eclext_FR.py (hunk 1 above)
Comment (flacombe) [reply analysers/analyser_merge_eclext_FR.py]: Should we create an issue for that?

**Comment by flacombe** [comment]
@frodrigo could you give me proper ids and classes for that please?

File: analysers/Analyser_Merge_street_lamp_FR_eclext.py (hunk 6 above)
Comment (flacombe) [reply analysers/Analyser_Merge_street_lamp_FR_eclext.py]: Done, hope it'll work

**Comment by frodrigo** [comment]
Thank you. Fixed and merged manually.

**Comment by frodrigo** [comment]
@flacombe it is deployed, but it fails with

```
'light:method': lambda res: self.extract_light_method[res['typeSource']] if self.extract_light_method[res.get('typeSource')] else None,
KeyError: 'AU'
```

**Pull request closed without merging**